maven/src/test/java/org/commonjava/maven/galley/maven/model/view/MavenPomViewTest.java

```diff
@@ -395,20 +395,28 @@ public void repositoriesWithUrlPropertyInProfile()
         MavenPomView pomView = loadPoms( "pom-with-repo-property-in-profile.xml" );
         List<RepositoryView> rvs = pomView.getAllRepositories();
 
-        for ( RepositoryView rv : rvs )
-        {
-            if ( rv.getName().equals( "repo.one" ) )
-            {
-                assertThat( rv.getUrl(), equalTo( "http://www.bar.com/repo" ) );
-            }
-            if ( rv.getName().equals( "test.oracle" ) )
-            {
-                assertThat( rv.getUrl(), equalTo( "http://test.oracle.repository" ) );
-            }
-            if ( rv.getName().equals( "test.two.oracle" ) )
-            {
-                assertThat( rv.getUrl(), equalTo( "http://test.two.oracle.repository" ) );
-            }
-        }
+        assertThat( rvs.get( 0 ).getName(), equalTo( "repo.one" ) );
+        assertThat( rvs.get( 0 ).getUrl(), equalTo( "http://repo.one.repository" ) );
+        assertThat( rvs.get( 1 ).getName(), equalTo( "test.oracle.repo" ) );
+        assertThat( rvs.get( 1 ).getUrl(), equalTo( "http://test.oracle.repository" ) );
+        assertThat( rvs.get( 2 ).getName(), equalTo( "test.second.oracle.repo" ) );
+        assertThat( rvs.get( 2 ).getUrl(), equalTo( "http://another.test.two.oracle.repository" ) );
+    }
+
+    @Test
+    public void resolveExpressionsBothInDepAndProfile()
```

Comment:
Do you know if this failed before your changes to RecursionInterceptor usage? It's a little different from how I would have implemented this test, but I think it should verify the things we care about. I was wondering if it failed before, just in case my code reading missed something...